lms/djangoapps/teams/csv.py

```diff
@@ -306,6 +310,20 @@ def validate_user_assignment_to_team_and_teamset(self, row):
                 return False
         return True
 
+    def validate_compatible_enrollment_modes(self, user, team_name, teamset_id):
+        """
+        Validates that only students enrolled in a masters track are on a single team. Disallows mixing of masters
+        with other enrollment modes on a single team.
+        """
+        if(teamset_id, team_name) not in self.user_enrollment_by_team:
+            self.user_enrollment_by_team[teamset_id, team_name] = set()
+        self.user_enrollment_by_team[teamset_id, team_name].add(self.user_to_actual_enrollment_mode[user.id])
+        if 'masters' in self.user_enrollment_by_team[teamset_id, team_name] and \
+                len(self.user_enrollment_by_team[teamset_id, team_name]) > 1:
+            self.add_error_and_check_if_max_exceeded('Masters track students can not be combined with other tracks.')
```

Comment:
Does @sapanathomas523 have thoughts on error messages that will be directly shown on the UI?

Reply:
Also, maybe add some more info? What team / what the violating user was? I guess given in the contect of the code there's no "violating user", it's just that the enrollment modes are mixed, but maybe if we get to this case we can find violating users? If it's an org_protected team we can list the 'proposed' non-masters users and visa versa?

Reply:
I posed the question to Sapana in the linked Jira. I remember we specifically wanted to have some messages be less obvious than others.